changelogs/unreleased/5672-therealak12-minor.md

```diff
@@ -0,0 +1,4 @@
+## Contour now waits for the cache sync before starting the DAG rebuild and XDS server
+
+Contour will wait for the initial list of Kubernetes objects to be processed and cached and then starts building its DAG and
```

Comment:
Maybe mention that this is change from previous behavior that just waited for informer caches to be synced but *not* for all the events to be fully processed that were an output of that sync, but now we use the client-go/controller-runtime mechanisms for doing so.

Reply:
Thanks. I've updated the release note.